site/docs/patterns/index.mdx

```diff
@@ -0,0 +1,524 @@
+---
+title: Forms
+layout: DetailTechnical
+sidebar:
+  label: Forms
+label: Forms
+---
+
+Forms allow you to capture data, gather opinions and receive information in a consistent format.
+
+This pattern provides practical and visually enriched guidelines for creating forms that are flexible to business domain, use experience, and channel.
+
+## Form layout
+
+Form controls align to a column layout with an even number of columns across breakpoints.
+
+<ImageSwitcher
+  images={[
+    { src: "/img/patterns/forms/form_layout_1.png", alt: "" },
+    { src: "/img/patterns/forms/form_layout_2.png", alt: "" },
+  ]}
+  label="Show spacing"
+/>
+
+\*You can find more information on the Salt spacing system here.
+
+<ExampleContainer type="positive">
+- Align the width of a Form Field to the number of columns that best reflect the length of its expected text.
+- Use the Figma "Salt Layout Grid" style library, which aligns to Salt Breakpoints. Simply pick the style that fits to your density and frame width.
+- Use "Compact layout" for very data-dense forms.
+</ExampleContainer>
+
+## Form container
+
+<Diagrams>
```

Comment:
Do we really need new `<Diagram(s)>` components for this?

Could we not achieve the same thing with existing stuff. For example:

```tsx
<GridLayout ...>
  <GridItem>

    ### Full page

    Embed your form...

    <Image ... />

  <GridItem>
 
  <GridItem>
    ...
  </GridItem>
</GridLayout>
```

That way we have less site-specific stuff to maintain going forwards.

Reply:
This would need a lot of custom CSS to work which is what I was trying to avoid. I think for now getting the patterns up is more important than trying to be DRY